Add option to control test repeat execution strategy

zloit opened the pull request:
This PR introduces a new configuration option parallelRepeats that gives users control over how repeated tests are executed:

**New features**
    - Add parallelRepeats option (default: true)
      Parallel mode (true): All test repeats are queued upfront for maximum throughput
      Sequential mode (false): Each repeat is queued only after the previous one completes

**Usage example**
```js
// In testplane config
plugins: {
  '@testplane/test-repeater': {
    repeat: 5,
    parallelRepeats: false // run repeats sequentially
  }
}
```

Commit:
feat: add strategy for controlling test repeat execution mode

Add 'parallelRepeats' option to control how test repeats are queued:
- When true (default): all repeats are queued immediately for parallel execution
- When false: repeats are queued one by one after each test completion

zloit commented on test/lib/repeater.js:
```diff
@@ -23,7 +23,7 @@ describe('Repeater', () => {
             it(`should not repeat ${name} test`, () => {
                 const repeatCounter = _mkRepeatCounter();
-                const repeater = Repeater.create(testplane, repeatCounter);
+                const repeater = Repeater.create(testplane, repeatCounter, true);
```
true by default

DudaGod commented on lib/repeat-counter.js:
```diff
@@ -21,6 +23,9 @@ module.exports = class RepeatCounter {
         if (repeatsLeft > 0) {
             logger.info(`Will be repeated. Repeats left: ${repeatsLeft}`);
+            if (!this._parallelRepeats) {
+                this._testplane.addTestToRun(test, test.browserId);
```
`repeatCounter` should not be responsible for repeating tests. It is just a counter.

We should implement additional method in `Repeater` which will be called if there are additional repeats for ended test.

DudaGod reviewed (commented):
Looks like that everything works fine, but to my mind `RepeatCounter` class shouldn't be responsible for tests repeat. We should just implement new method in `Counter` class for this purpose. Can we fix it? Thank you for your contribution.

Commit:
refactor: separate responsibilities between RepeatCounter and Repeater

- RepeatCounter is now only responsible for counting repeats
- Repeater handles test repetition logic via new handleTestEnd() method
- Updated documentation for parallelRepeats parameter in both languages
- Added comprehensive tests for new functionality
- All tests passing

Fixes review feedback about RepeatCounter responsibilities

DudaGod reviewed (approved):
:fire:
